### PR TITLE
Metrics included/excluded categories

### PR DESCRIPTION
**Pull request by ashleyzhang**
Screen Shot 2022-04-26 at 5 56 49 PM

## Description
Add a configurations tab to metrics feature that includes a "Considered Categories" section enabling professors to decide what assessment categories are or are not included in watchlist calculations.

## Motivation and Context
Some courses have optional assessments that do not factor into determining whether or not a student is at risk. Allowing instructors to exclude a category of assessments reduces false positive rates for students appearing in the watchlist.

## How Has This Been Tested?
Navigate to the configurations tab in the student metrics dashboard. Exclude a category in the watchlist, and save the changes. Ensure that the change is reflected in the watchlist. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have run rubocop for style check. If you haven't, run `overcommit --install && overcommit --sign` to use pre-commit hook for linting
- [x] My change requires a change to the documentation, which is located at [Autolab Docs](https://github.com/autolab/docs)
- [ ] I have updated the documentation accordingly, included in this PR

## Other issues / help required

If unsure, feel free to submit first and we'll help you along.

### CODE REVIEW

**Comment by victorhuangwq** [comment]
Tested. It seems like the default behavior for when you resolve and then change the configuration is to archive everything (just like a change of metrics). Is that intended behavior?

File: db/schema.rb [hunk 1]
```diff
@@ -242,11 +244,12 @@
     t.datetime "resolved_at"
     t.integer "resource_owner_id"
     t.string "access_code"
+    t.index ["application_id"], name: "fk_rails_4035c6e0ed"
```
Comment (victorhuangwq) [inline db/schema.rb:247]: this seems to be a bit odd to push up @ashleyzhang, as this looks like a config?

**Comment by xinyis991105** [comment]
> Tested. It seems like the default behavior for when you resolve and then change the configuration is to archive everything (just like a change of metrics). Is that intended behavior?

Yes this is expected! Currently a change in metrics config has the same effect on watchlist instances as a change in risk conditions, i.e. refreshing the list, archiving old instances.

**Comment by xinyis991105** [comment]
In addition, I've had some misunderstanding of the use of the word "allowlist" and currently allowlist contains all the categories that are not considered in the calculation of metrics. Do we want to change this to blocklist afterwards?

**Comment by victorhuangwq** [comment]
I think that would be confusing - would it be possible to change that in this PR? I'm afraid we might forget that later

**Comment by xinyis991105** [comment]
> I think that would be confusing - would it be possible to change that in this PR? I'm afraid we might forget that later

See the latest changes. I rollbacked and changed the relevant migration file instead of writing another migration file for renaming.

**Review by victorhuangwq — APPROVED** [review]
Did a rails db:reset and utilized the new schema. Feature works as expected. Refactor looks fine.
LGTM!

**Pull request merged**